<?php

use Illuminate\Foundation\Application;
use Illuminate\Foundation\Configuration\Exceptions;
use Illuminate\Foundation\Configuration\Middleware;
use Illuminate\Validation\ValidationException;
use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
use App\Http\Middleware\JwtMiddleware;

return Application::configure(basePath: dirname(__DIR__))
  ->withRouting(web: __DIR__ . '/../routes/web.php', api: __DIR__ . '/../routes/api.php', commands: __DIR__ . '/../routes/console.php', health: '/up')
  ->withMiddleware(function (Middleware $middleware): void {
    $middleware->alias([
      'jwt.auth' => JwtMiddleware::class,
    ]);
  })
  ->withExceptions(function (Exceptions $exceptions): void {
    $exceptions->render(function (Throwable $e, $request) {
      if ($e instanceof ValidationException) {
        return response()->json(
          [
            'message' => $e->getMessage(),
            'statusCode' => 422,
          ],
          422,
        );
      }
      if ($e instanceof HttpExceptionInterface) {
        return response()->json(
          [
            'message' => $e->getMessage(),
            'statusCode' => $e->getStatusCode(),
          ],
          $e->getStatusCode(),
        );
      }
      return response()->json(
        [
          'message' => $e->getMessage(),
          'statusCode' => 500,
        ],
        500,
      );
    });
  })
  ->create();
